File: app/src/at-tamle/java/asiantech/internship/winter/baitap/bai8.java
package asiantech.internship.winter.baitap;
import java.util.Scanner;
//liet ke all so thuan nghich 6 chu so
public class bai8 {
	public static boolean testSoThuanNghich(int n) {
		  StringBuilder xau = new StringBuilder();
		  String str = "" + n;
		  xau.append(str);
		  String check = "" + xau.reverse();
		  if (str.equals(check))
		   return true;
		  else
		   return false;
		 }

	 public static void main(String[] args) {
		  int n, count = 0;
		  for (n = 100000; n <= 999999; n++) {
		   if (testSoThuanNghich(n)) {
		    System.out.println(n);
		    count++;
		   }
		  }
		  System.out.println("Co " + count + " so thuan nghich co 6 chu so");
		 }

}

File: app/src/at-trinhnguyen/java/asiantech/internship/winter/javacore/Bai8.kt
package asiantech.internship.winter.javacore

import java.util.*

/*
Bài 8. Một số được gọi là số thuận nghịch độc nếu ta đọc
 từ trái sang phải hay từ phải sang trái số đó ta vẫn nhận được
  một số giống nhau. Hãy liệt kê tất cả các số thuận nghịch độc
  có sáu chữ số (Ví dụ số: 558855).
 */
object Bai8 {
    @JvmStatic
    fun main(args: Array<String>) {
        val scanner = Scanner(System.`in`)
        print("Enter n: ")
        val n = scanner.nextInt()
        if (isReversible(n.toString())) {
            print("$n is reversible")
        } else print("$n is not reversible")

    }

    private fun isReversible(n: String): Boolean {
        return n.equals(StringBuffer(n).reverse().toString())
    }
}